plugins/woocommerce/src/Admin/API/Options.php

```diff
@@ -99,6 +107,11 @@ public function user_has_permission( $option, $request, $is_update = false ) {
 			return $permissions[ $option ];
 		}
 
+		// Don't allow to update options in non-production environments if the option is not whitelisted. This is to force developers to update the option permissions when adding new options.
+		if ( 'production' !== wp_get_environment_type() ) {
+			return false;
+		}
+
 		wc_deprecated_function( 'Automattic\WooCommerce\Admin\API\Options::' . ( $is_update ? 'update_options' : 'get_options' ), '3.1' );
```

Comment:
I wonder if it's worth updating the version reference `3.1` here? It used to refer to WooCommerce Admin version 3.1. I believe it's paired with [WooCommerce 6.3](https://developer.woocommerce.com/2022/03/08/woocommerce-6-3-released/)

Reply:
Yeah, good catch. Updated in 2bd30cafc9f540161d4659e1311507b7f8069f5c.

Could you please approve this PR again? Thank you! 😄